backend/Watchdog.Core/Watchdog.Core.DAL/Context/EntityConfigurations/ApplicationConfig.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Watchdog.Core.DAL.Entities;
+
+namespace Watchdog.Core.DAL.Context.EntityConfigurations
+{
+    public class ApplicationConfig : IEntityTypeConfiguration<Application>
+    {
+        public void Configure(EntityTypeBuilder<Application> builder)
+        {
+            builder.Property(e => e.CreatedBy)
+                  .IsRequired();
+
+            builder.Property(e => e.CreatedAt)
+                  .IsRequired();
+
+            builder.Property(e => e.Name)
+                  .HasMaxLength(128)
+                  .IsRequired();
+
+            builder.Property(e => e.SecurityToken)
+                  .HasMaxLength(256)
+                  .IsRequired();
+
+            builder.HasMany(e => e.Environments)
+                  .WithOne(e => e.Application)
```

Comment:
If the Application is optional then make ApplicationId as a nullable. It cannot be optional and has required (int) FK property. So you don't need IsRequired for navigation properties, but it would be good to set HasForeignKey for all relations.

Reply:
I explicitly specified foreign keys for all relationships and made the optional properties nullable.